File: 1512-number-of-good-pairs/1512-number-of-good-pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        map<int,int> mp;
        
        for(int i : nums){
            mp[i]++;
        }
        int ans = 0;
        for(auto it : mp){
            int n = it.second;
            int add = n*(n-1)/2;
            ans += add;
        }
        
        return ans;
    }
};

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.go
func numIdenticalPairs(nums []int) int {
    n := len(nums)
    ans := 0
    mp := make(map[int]int)
    
    for i := 0; i < n; i++ {
        ans += mp[nums[i]]
        mp[nums[i]]++
    }
    return ans
}